components/feature_restricted_modal/feature_restricted_modal.tsx

```diff
@@ -112,6 +125,26 @@ const FeatureRestrictedModal = ({
         secondaryBtnAction = customSecondaryButton.action;
     }
 
+    let trialBtn;
```

Comment:
to avoid the if...else... you can initialize the trialBtn with the selfhosted StartTrialBtn component, and only if cloud, you set the CloudStartTrialButton